presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -608,6 +609,17 @@ public static NullableValue parsePartitionValue(HivePartitionKey key, Type type,
         return parsePartitionValue(key.getName(), key.getValue().orElse(HIVE_DEFAULT_DYNAMIC_PARTITION), type, timeZone);
     }
 
+    public static NullableValue parsePartitionValue(String partitionName, String value, Type type, ZoneId hiveStorageTimeZoneId)
+    {
+        return parsePartitionValue(partitionName, value, type, getDateTimeZone(hiveStorageTimeZoneId));
+    }
+
+    private static DateTimeZone getDateTimeZone(ZoneId hiveStorageTimeZoneId)
+    {
+        requireNonNull(hiveStorageTimeZoneId, "hiveStorageTimeZoneId is null");
+        return DateTimeZone.forID(hiveStorageTimeZoneId.getId());
```

Comment:
Could not add `DateTimeZone` at the call site because `joda-time` was replaced by `java.time` in https://github.com/prestodb/presto/commit/bc82daf13276c44547084eae83671e7eb285f4ee 
So, i have added this util method.

Reply:
We could do as you suggested but it will entail copying significant code from HiveUtil, which also contains some usage of deprecated api. I think it's better for maintainability to do it the current way.

Added a test in `TestHiveUtil`.

Reply:
>but it will entail copying significant code from HiveUtil

How so? All the methods you call here like `parsePartitionValue`  are public static right?

Reply:
Yeah all relevant methods in this call are public static but they work with `org.joda.time.DateTimeZone`. In order to avoid adding that library in presto-hudi module, i'll have to copy and rewrite those methods to make them work with `java.time.ZoneId`.